File: tmdb-core/src/main/java/dev/reuss/tmdb/common/image/PersonImages.java
package dev.reuss.tmdb.common.image;

import com.fasterxml.jackson.annotation.JsonIgnoreProperties;
import dev.reuss.tmdb.common.TmdbModel;
import dev.reuss.tmdb.util.TmdbCollections;

import java.util.List;

/**
 * Image metadata for a TMDB person.
 *
 * @param id       TMDB person id
 * @param profiles profile images
 */
@JsonIgnoreProperties(ignoreUnknown = true)
public record PersonImages(
        int id,
        List<TmdbImage> profiles
) implements TmdbModel {

    public PersonImages {
        profiles = TmdbCollections.list(profiles);
    }
}

File: tmdb-core/src/main/kotlin/dev/reuss/tmdb/common/image/PersonImages.kt
package dev.reuss.tmdb.common.image

import com.fasterxml.jackson.annotation.JsonProperty
import dev.reuss.tmdb.common.TmdbModel

/**
 * Image metadata for a TMDB person.
 *
 * @property id       TMDB person id
 * @property profiles profile images
 */
@JvmRecord
data class PersonImages(
    @all:JsonProperty("id")
    val id: Int,
    @all:JsonProperty("profiles")
    val profiles: List<TmdbImage> = emptyList(),
) : TmdbModel
